Tensile/TensileBenchmarkCluster.py

```diff
@@ -0,0 +1,846 @@
+        sBuffer.writeLine("[ -f \"${taskScript}\" ] || failAndExit \"Task script not found\";")
+        sBuffer.writeEcho("Image dir: $imageDir")
+        sBuffer.writeEcho("Logs dir: $logsDir")
+        sBuffer.writeEcho("Tasks dir: $tasksDir")
+        sBuffer.writeEcho("Results dir: $resultsDir")
+        sBuffer.writeEcho("Enqueue script: $taskScript")
+        sBuffer.writeLine("")
+
+    @staticmethod
+    def __configureJob(sBuffer):
+        sBuffer.writeComment("Setup local environment")
+        sBuffer.writeComment("Get in the directory where this script lives")
+        sBuffer.writeLine("pushd $( dirname `greadlink -f ${BASH_SOURCE[0]} || readlink -f ${BASH_SOURCE[0]}` )")
+        sBuffer.writeComment("Enumerate the tasks and set up batch params based on task count")
+        sBuffer.writeLine("subDirs=($tasksDir/*)")
+        sBuffer.writeLine("let arraySize=${#subDirs[@]}")
+        sBuffer.writeLine("[ -z \"${arraySize}\" ] && failAndExit \"No task count\";")
+        sBuffer.writeLine("let arrayStart=0")
+        sBuffer.writeLine("let arrayEnd=$arraySize-1")
+        sBuffer.writeComment("Create logs directory for SLURM")
+        sBuffer.writeLine("slurmLogsDir=$logsDir/SLURM")
+        sBuffer.writeLine("mkdir -p $slurmLogsDir")
+        sBuffer.writeLine("")
+
+    @staticmethod
+    def __submitJob(sBuffer):
+        sBuffer.writeComment("Invoke batch request to SLURM")
+        sBuffer.writeComment("Will wait until the task completes before exiting")
+        sBuffer.writeLine("runCmd=\"sbatch --nodes=1 --array=$arrayStart-$arrayEnd -o $slurmLogsDir/slurm-%A_%a.out --wait $taskScript -i $imageDir -l $logsDir -r $resultsDir -t $tasksDir\"")
+        sBuffer.writeEcho("Submitting batch job to SLURM. This may take a while...")
+        sBuffer.writeEcho("$runCmd")
+        sBuffer.writeLine("$runCmd")
+        sBuffer.writeLine("returnCode=$?")
+        sBuffer.writeLine("popd")
+
+    @staticmethod
+    def __exit(sBuffer):
+        sBuffer.writeEcho("Done!")
+        sBuffer.writeLine("exit $returnCode")
+
+    @staticmethod
+    def writeHeader(sBuffer):
+        BenchmarkJobWriter.__initHeader(sBuffer)
+
+    @staticmethod
+    def writeBody(sBuffer):
+        BenchmarkJobWriter.__initializeFuncs(sBuffer)
+        BenchmarkJobWriter.__parseArgs(sBuffer)
+        BenchmarkJobWriter.__configureJob(sBuffer)
+        BenchmarkJobWriter.__submitJob(sBuffer)
+
+    @staticmethod
+    def writeEpilogue(sBuffer):
+        BenchmarkJobWriter.__exit(sBuffer)
+
+class BenchmarkImplSLURM(object):
+
+    # baseImage: remote URL to rocm build
+    # dockerFilePath: path to docker file for slurm tensile build
+    # tag: name:TAG of new image
+    # outDir: where to export the new image
+    # tensile<Fork/Branch/Commit>: tensile code to use on github
+    @staticmethod
+    def __createTensileBenchmarkContainer(baseImage, dockerFilePath, tag, outDir, logDir, tensileFork, tensileBranch, tensileCommit):
```

Comment:
There's considerable effort here to abstract away Slurm but Docker is implicitly used everywhere.  Getting images from one machine to another and starting containers is often handled implicitly by the scheduler.